### DISCUSSION TITLE
Are we being TOO draconian, about this "only support the latest GNOME" thing?

### DISCUSSION THREAD

**Discussion by ferdnyc** [andyholmes/gnome-shell-extension-gsconnect#1612] [Ideas]
### Describe your request

I get the theory behind only supporting the latest version, but I'm not sure I get the practice.

For example, currently we "only support GNOME 44". Except, there are no distros _out there_ running GNOME 44 yet. Or at least, there weren't when we made the move to GNOME 44. Fedora 38 is still a couple of weeks from release, even now. If Ubuntu 23.4 is out, it's only recently released. But we've been GNOME 44-only since #1577 was merged on March 13, when there were _definitely_ no distros supporting GNOME 44.

And most of the backwards-incompatible changes aren't **necessary**. Like, the one that broke GNOME 43 compatibility was this change in the QuickMenuToggle `_init`:

```diff
    _init() {
        super._init({
-            label: 'GSConnect',
-            iconName: 'org.gnome.Shell.Extensions.GSConnect-symbolic',
+            title: 'GSConnect',
```

But here's the thing: `label` is _deprecated_ in that function, but it's not _broken_. If we'd left it using `label` instead, the extension would still be GNOME 43 compatible, _and_ work under GNOME 44. Why are we deliberately breaking things for users?

### Proposed solution

Don't deliberately (and unnecessarily) break things for users?

IOW, keep compatibility _where possible_, even if it means staying with deprecated (but still fully functional) API on the latest-and-greatest version for now. We can always create Projects reminders to update things down the road.

### Alternatives

For users of GNOME 43 distros? Not many.

* Just keep telling them to live with bugs until they upgrade, I guess.
* Start keeping branches for multiple versions? I'd really like to avoid that if possible.

### GSConnect version

55

### Installed from

GitHub releases (zip file)

### GNOME Shell version

Any version before 44

### Linux distribution/release

Basically all, at the time #1577 was merged

**Comment by ferdnyc** [comment]
On further consideration, this should be in Discussions. Moving.

**Reply by ferdnyc** [reply]
> Fedora 38 is still a couple of weeks from release, even now.

Shows what I know. It actually released last week, a week early.

**Reply by daniellandau** [reply]
Ubuntu too

**Comment by daniellandau** [comment]
We are keeping branches for older shell versions. I released patch versions for multiple older shell versions for a memory leak a while back and then a distro package ook the largest number "latest" release that was the oldest gsconnect for the oldest shell. It was a whole season of very unnecessary bug reports.

That said, I think we still should backport fixes to old releases in the future as well when the bug being fixed is major enough. Should probably order the releases in another order and be clearer in the release notes in the hopes that it helps anything.

I don't want to have a single code base claiming support for many versions, as then I'd have to test with all of those versions for each release.

**Reply by ferdnyc** [reply]
Mmm, I guess the thing we missed, when merging the GNOME 44 update, was creating a `gnome-43` branch from the commit prior to that. We have branches for _even older_ GNOME versions (something I hadn't realized, TBH), but not the one that's — at least until very recently / for a little while longer — currently in use by the majority of distros, and for which the majority of bug reports are still likely to be coming in.

**Reply by daniellandau** [reply]
Yeah. Creating the branch pre-emptively might theoretically be more welcoming to contributors, but in reality I'd think they'd step up in a discussion on an issue first. And if it's me that would do the fix to an old release I can just create the branch in the moment.

**Comment by andyholmes** [comment]
I'd generally agree with Daniel.

The two factors here are really time and testing. Personally, I just don't have time to run multiple versions of GNOME on multiple distributions (and distro does make a difference here too). Supporting multiple versions in the same release is not a fun game to play, I can tell you that.

The only reasonable way is to keep separate branches, in my experience. Ideally, we could just branch before a new GNOME release and if it's *actually* used someone will be running it and backporting or requesting backports.

I guess my opinion is still generally: if someone wants to do the work, it's fine by me :shrug:
